File: src/main/java/de/bhclub/ticker/text/ScrollingTextService.java
package de.bhclub.ticker.text;

import de.bhclub.ticker.playlists.PlaylistNotFoundException;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.awt.*;
import java.util.Optional;

@Service
public class ScrollingTextService {

    @Autowired
    private ScrollingTextRepository scrollingTextRepository;

    public void createText(String text, String textColor, String outline, String background, String font, String speed, String spacing) {
        ScrollingText scrollingText = new ScrollingText();
        scrollingText.setText(text);
        scrollingText.setTextColor(Color.decode(textColor));
        scrollingText.setOutlineColor(Color.decode(outline));
        scrollingText.setBackground(Color.decode(background));
        scrollingText.setFont(font);
        scrollingText.setSpeed(Integer.parseInt(speed));
        scrollingText.setSpacing(Integer.parseInt(spacing));

        scrollingTextRepository.save(scrollingText);
    }

    public Iterable<ScrollingText> getAll() {
        return scrollingTextRepository.findAll();
    }

    public ScrollingText getText(Long id) {
        Optional<ScrollingText> text = scrollingTextRepository.findById(id);
        if (!text.isPresent()) {
            throw new PlaylistNotFoundException("text not found");
        }
        return text.get();
    }

    public static String toHex(Color color) {
        return String.format("#%06X", (0xFFFFFF & color.getRGB()));
    }

    public void deleteText(Long id) {
        scrollingTextRepository.deleteById(id);
    }
}

File: src/main/kotlin/de/bhclub/ticker/text/ScrollingTextService.kt
package de.bhclub.ticker.text

import de.bhclub.ticker.playlists.PlaylistNotFoundException
import org.springframework.stereotype.Service
import java.awt.Color
import java.util.UUID

@Service
class ScrollingTextService(
    private val scrollingTextRepository: ScrollingTextRepository,
) {
    fun createText(
        text: String,
        textColor: String,
        outline: String,
        background: String,
        font: String,
        speed: String,
        spacing: String
    ) {
        val scrollingText = ScrollingText(
            text = text,
            textColor = Color.decode(textColor),
            outlineColor = Color.decode(outline),
            background = Color.decode(background),
            font = font,
            speed = speed.toInt(),
            spacing = spacing.toInt(),
        )
        scrollingTextRepository.save(scrollingText)
    }

    fun getAll() = scrollingTextRepository.findAll()

    fun getText(id: UUID): ScrollingText {
        val text = scrollingTextRepository.findById(id)
        if (!text.isPresent) {
            throw PlaylistNotFoundException("text not found")
        }
        return text.get()
    }

    fun deleteText(id: UUID) {
        scrollingTextRepository.deleteById(id)
    }

    companion object {
        @JvmStatic
        fun toHex(color: Color): String {
            return String.format("#%06X", 0xFFFFFF and color.rgb)
        }
    }
}